src/OpenBinary/reader.jl

```diff
@@ -156,137 +147,157 @@ function PSRI.open(
         _first_stage = read(ioh, Int32) #month or week
         first_year = read(ioh, Int32) # year
         #
-        unit_buffer = Vector{Cchar}(undef, 7)
-        read!(ioh, unit_buffer)
-        unit_str = strip(join(Char.(unit_buffer)))
+        unit_str = strip(join(Char.(read!(ioh, Vector{Cchar}(undef, 7)))))
         #
         name_length = read(ioh, Int32)
 
         hour_discretization = 1
 
         if variable_by_hour == 0
-            skip_store(ioh, 4, skips, store_skips)
-            skip_store(ioh, 4, skips, store_skips)
+            skip(ioh, 4)
+            skip(ioh, 4)
+
             offset1 = read(ioh, Int32)
             offset2 = read(ioh, Int32)
+            
             block_total = offset2 - offset1
-            skip_store(ioh, 4 * (stage_total - first_relative_stage), skips, store_skips)
+
+            skip(ioh, 4 * (stage_total - first_relative_stage))
+            
             number_blocks = Int32[]
         elseif variable_by_hour == 1
-            skip_store(ioh, 4, skips, store_skips)
-            skip_store(ioh, 4, skips, store_skips)
+            skip(ioh, 4)
+            skip(ioh, 4)
+
             number_blocks = zeros(Int32, stage_total + 1 - first_relative_stage)
             offsets = zeros(Int32, 1 + stage_total + 1 - first_relative_stage)
-            for i in first_relative_stage:stage_total+1
-                offset = read(ioh, Int32)
-                offsets[i-first_relative_stage+1] = offset
-                if i > first_relative_stage
-                    number_blocks[i-first_relative_stage] = offset - offsets[i-first_relative_stage]
+            
+            for i in first_relative_stage:(stage_total+1)
+                let offset = read(ioh, Int32)
```

Comment:
Why do you need a let statement here? Maybe consider renaming these offsets to some more informative name such as `header_offset`, `stage_offset`, etc.